include/irr/asset/IDescriptorSetLayout.h

```diff
@@ -24,20 +25,62 @@ enum E_DESCRIPTOR_TYPE
 
 class IDescriptorSetLayout
 {
-public:
-    struct SBinding
+    template<typename SamplersType>
+    struct SBinding_
     {
         E_DESCRIPTOR_TYPE type;
         uint32_t count;
         E_SHADER_STAGE stageFlags;
-        core::smart_refctd_dynamic_array<SSamplerParams> samplers;
+        SamplersType samplers;
     };
+    using SBinding_internal = SBinding_<const SSamplerParams*>;
+public:
+    using SBinding = SBinding_<core::smart_refctd_dynamic_array<SSamplerParams>>;
```

Comment:
you don't need a refcounted array here, just let people set it to plain pointers. You won't be holding that array after the function returns.

Only use refcounted pointers if you will hold the pointee after function returns.

Reply:
hm, right, thanks

Reply:
lemme know when changed.

Reply:
resolved